selfdrive/car/subaru/carstate.py

```diff
@@ -102,64 +115,110 @@ def get_can_parser(CP):
+        ("CruiseControl", 20),
+      ]
+
     return CANParser(DBC[CP.carFingerprint]['pt'], signals, checks, 0)
 
   @staticmethod
   def get_cam_can_parser(CP):
-    signals = [
-      ("Cruise_Set_Speed", "ES_DashStatus", 0),
-      ("Conventional_Cruise", "ES_DashStatus", 0),
-
-      ("Counter", "ES_Distance", 0),
-      ("Signal1", "ES_Distance", 0),
-      ("Cruise_Fault", "ES_Distance", 0),
-      ("Cruise_Throttle", "ES_Distance", 0),
-      ("Signal2", "ES_Distance", 0),
-      ("Car_Follow", "ES_Distance", 0),
-      ("Signal3", "ES_Distance", 0),
-      ("Cruise_Brake_Active", "ES_Distance", 0),
-      ("Distance_Swap", "ES_Distance", 0),
-      ("Cruise_EPB", "ES_Distance", 0),
-      ("Signal4", "ES_Distance", 0),
-      ("Close_Distance", "ES_Distance", 0),
-      ("Signal5", "ES_Distance", 0),
-      ("Cruise_Cancel", "ES_Distance", 0),
-      ("Cruise_Set", "ES_Distance", 0),
-      ("Cruise_Resume", "ES_Distance", 0),
-      ("Signal6", "ES_Distance", 0),
-
-      ("Counter", "ES_LKAS_State", 0),
-      ("Keep_Hands_On_Wheel", "ES_LKAS_State", 0),
-      ("Empty_Box", "ES_LKAS_State", 0),
-      ("Signal1", "ES_LKAS_State", 0),
-      ("LKAS_ACTIVE", "ES_LKAS_State", 0),
-      ("Signal2", "ES_LKAS_State", 0),
-      ("Backward_Speed_Limit_Menu", "ES_LKAS_State", 0),
-      ("LKAS_ENABLE_3", "ES_LKAS_State", 0),
-      ("LKAS_Left_Line_Light_Blink", "ES_LKAS_State", 0),
-      ("LKAS_ENABLE_2", "ES_LKAS_State", 0),
-      ("LKAS_Right_Line_Light_Blink", "ES_LKAS_State", 0),
-      ("LKAS_Left_Line_Visible", "ES_LKAS_State", 0),
-      ("LKAS_Left_Line_Green", "ES_LKAS_State", 0),
-      ("LKAS_Right_Line_Visible", "ES_LKAS_State", 0),
-      ("LKAS_Right_Line_Green", "ES_LKAS_State", 0),
-      ("LKAS_Alert", "ES_LKAS_State", 0),
-      ("Signal3", "ES_LKAS_State", 0),
-    ]
-
-    checks = [
-      ("ES_DashStatus", 10),
-      ("ES_Distance", 20),
-      ("ES_LKAS_State", 10),
-    ]
+    if CP.carFingerprint in PREGLOBAL_CARS:
+      signals = [
+        ("Cruise_Set_Speed", "ES_DashStatus", 0),
+        ("Not_Ready_Startup", "ES_DashStatus", 0),
+
+        ("Throttle_Cruise", "ES_CruiseThrottle", 0),
+        ("Signal1", "ES_CruiseThrottle", 0),
+        ("Cruise_Activated", "ES_CruiseThrottle", 0),
+        ("Signal2", "ES_CruiseThrottle", 0),
+        ("Brake_On", "ES_CruiseThrottle", 0),
+        ("DistanceSwap", "ES_CruiseThrottle", 0),
+        ("Standstill", "ES_CruiseThrottle", 0),
+        ("Signal3", "ES_CruiseThrottle", 0),
+        ("CloseDistance", "ES_CruiseThrottle", 0),
+        ("Signal4", "ES_CruiseThrottle", 0),
+        ("Standstill_2", "ES_CruiseThrottle", 0),
+        ("ES_Error", "ES_CruiseThrottle", 0),
+        ("Signal5", "ES_CruiseThrottle", 0),
+        ("Counter", "ES_CruiseThrottle", 0),
+        ("Signal6", "ES_CruiseThrottle", 0),
+        ("Button", "ES_CruiseThrottle", 0),
+        ("Signal7", "ES_CruiseThrottle", 0),
+      ]
+
+      checks = [
+        ("ES_DashStatus", 20),
+      ]
```

Comment:
Add checks for all messages you're reading in carState

Reply:
done, I did not include BodyInfo and Dash_State since they are ~0.8Hz and I think 1Hz is minimum